feat: add option cwd

mojoaxel opened the pull request.

eddiemonge commented on package.json:
```diff
@@ -67,11 +67,11 @@
   "devDependencies": {
     "grunt": "^1.0.1",
-    "grunt-auto-release": "^0.0.6",
+    "grunt-auto-release": "^0.0.7",
```
can you move these to a new PR?

mojoaxel replied:
yeah. ok. it's cleaner that way.

eddiemonge commented on tasks/bump.js:
```diff
@@ -30,6 +33,20 @@ module.exports = function(grunt) {
 
+    // check if cwd exists
+    if (!fs.existsSync(opts.cwd)) {
+      grunt.fatal('The in "cwd" defined directory does not exist');
```
`The "cwd" requested does not exist`

mojoaxel replied:
:+1:

eddiemonge commented on tasks/bump.js:
```diff
@@ -30,6 +33,20 @@ module.exports = function(grunt) {
+    };
+
+    // set grunts default working directory
+    grunt.verbose.writeln('Changing working directory: ' + opts.cwd);
```
Only show this if cwd `!== '.'`

mojoaxel replied:
I think it doesn't hurt if the current working directory is shown to the user. This is only in `--verbose` mode. But I'm also ok, with the change.

eddiemonge commented on tasks/bump.js:
```diff
@@ -30,6 +33,20 @@ module.exports = function(grunt) {
+
+    // set grunts default working directory
+    grunt.verbose.writeln('Changing working directory: ' + opts.cwd);
+    grunt.file.setBase(opts.cwd);
```
I don't know if this sets this just for this task or all tasks so maybe store the current base and then restore it after the tasks have run.

mojoaxel replied:
Good point! I haven't tested it with multiple tasks. I better change it back!

mojoaxel committed:
feat: add option cwd

With the new option `cwd` it is possible to define a custom working directory.
This is usefull if you want to pump e.g a sub-component or simply for testing.

Closes #198

eddiemonge commented on tasks/bump.js:
```diff
@@ -60,6 +81,11 @@ module.exports = function(grunt) {
+
+        // reset original cwd
+        if (originalCwd) {
+          grunt.file.setBase(originalCwd);
```
wont this mess up the file list for every exec after the first? Should the original changeBase go in the runIf?

eddiemonge commented:
Closing due to inactivity
